src/vsr/client.zig

```diff
@@ -19,21 +19,130 @@ pub fn Client(comptime StateMachine_: type, comptime MessageBus: type) type {
         pub const StateMachine = StateMachine_;
 
         pub const Error = error{
+            BatchSizeZero,
```

Comment:
I think it should be possible more generally for an operation to allow an empty body. (Even though none of the operations besides `register` currently do so). Maybe `StateMachine` should have a set of `batch_min` constants (mirroring `batch_max`).

Reply:
In a hypothetical case of empty body, would it be correct to batch multiple logical batches into a single request? Since there is no message body, I think this particular case would require multiple physical requests as well to keep the intended semantics. Otherwise, the server will see only one message, not multiple.

Let's move all message body validations and the decision whether an operation can be batched or not to the `StateMachine`. 👍

Reply:
> Since there is no message body, I think this particular case would require multiple physical requests as well to keep the intended semantics.

Agreed. I think that an empty body should not be batched.

> Let's move all message body validations and the decision whether an operation can be batched or not to the StateMachine.

We do want to keep as much as possible out of the state machine, to make it easier for users to create custom state machines. If there's anything that we can possibly validate outside of the state machine, then we should do that, to keep complexity contained within VSR.